pkg/controller/account/account_controller.go

```diff
@@ -573,25 +573,34 @@ func CreateAccount(reqLogger logr.Logger, client awsclient.Client, accountName,
 	return accountStatus, nil
 }
 
-func checkAWSAccountsLimitReached(r *ReconcileAccount, reqLogger logr.Logger, currentAccounts int) (bool, error) {
-	instance := &corev1.ConfigMap{}
-	err := r.Client.Get(context.TODO(), types.NamespacedName{Namespace: awsv1alpha1.AccountCrNamespace, Name: awsv1alpha1.DefaultConfigMap}, instance)
+// accountLimitReached returns True if our account limit is reached or False if the account limit is not reached and we can create accounts.
+func accountLimitReached(kubeClient client.Client, log logr.Logger, currentAccounts int) bool {
+	limit, err := getAwsAccountLimit(kubeClient)
 	if err != nil {
-		unexpectedErrorMsg := fmt.Sprintf("%s: Failed to retrieve default ConfigMap, account limit defaulting to 100", awsv1alpha1.ErrMissingDefaultConfigMap)
-		reqLogger.Info(unexpectedErrorMsg)
-	} else {
-		if limit, ok := instance.Data["account-limit"]; ok {
-			if i, err := strconv.Atoi(limit); err == nil {
-				return i <= currentAccounts, nil
-			}
-			unexpectedErrorMsg := fmt.Sprintf("Account: Failed to convert ConfigMap 'account-limit' string field to int, account limit defaulting to 100")
-			reqLogger.Info(unexpectedErrorMsg)
-		} else {
-			unexpectedErrorMsg := fmt.Sprintf("%s: Default ConfigMap missing 'account-limit' field, account limit defaulting to 100", awsv1alpha1.ErrInvalidConfigMap)
-			reqLogger.Info(unexpectedErrorMsg)
-		}
+		log.Error(err, "There was an error getting the limits.  Using the default value.")
```

Comment:
I did a Tim Allen here, seeing this message declaring that we're using the default value, but having to look at the logic of the called func to see how that's being done. I stared at it for a while and I see there's just not a great way to organize this code. But it might read better if `getAwsAccountLimit` returned a deliberately bogus `limit` on errors (like `-1`??) and `limit = awsv1alpha1.DefaultConfigMapAccountLimit` happened here in this conditional.